angel_system/global_step_prediction/medical_domain_experiment.py

```diff
@@ -0,0 +1,159 @@
+        activity_confs = video_dset.images().lookup("activity_conf")
+        activity_gts = video_dset.images().lookup("activity_gt")
+
+        def get_unique(activity_ids):
+            """
+            Get unique list indexes without sorting the list.
+            """
+            indexes = np.unique(activity_ids, return_index=True)[1]
+            return [activity_ids[index] for index in sorted(indexes)]
+
+        print(f"unique activities: {get_unique(activity_gts)}")
+
+        step_predictor.process_new_confidences(activity_confs)
+
+        recipe_type = step_predictor.determine_recipe_from_gt_first_activity(
+            activity_gts
+        )
+        print(f"recipe_type = {recipe_type}")
+        if recipe_type == "unknown_recipe_type":
+            print("skipping plotting.")
+            continue
+        config_fn = config[recipe_type]
+        (
+            granular_step_gts,
+            granular_step_gts_no_background,
+            broad_step_gts,
+            broad_step_gts_no_background,
+        ) = step_predictor.get_gt_steps_from_gt_activities(video_dset, config_fn)
+
+        print(f"unique broad steps: {get_unique(broad_step_gts)}")
+
+        step_predictor.plot_gt_vs_predicted_one_recipe(
+            granular_step_gts,
+            recipe_type,
+            fname_suffix=f"{str(vid_id)}_granular_{extra_output_suffix}",
+            granular_or_broad="granular",
+        )
+        step_predictor.plot_gt_vs_predicted_one_recipe(
+            broad_step_gts,
+            recipe_type,
+            fname_suffix=f"{str(vid_id)}_broad_{extra_output_suffix}",
+            granular_or_broad="broad",
+        )
+
+
+if __name__ == "__main__":
+    """
+    coco_val = kwcoco.CocoDataset(
+        "/data/PTG/cooking/training/activity_classifier/TCN_HPL/logs/yolo_all_recipes_sample_rate_2/runs/2023-10-25_12-08-48/val_activity_preds_epoch69.mscoco.json"
+    )
+    coco_test = kwcoco.CocoDataset(
+        "/data/PTG/cooking/training/activity_classifier/TCN_HPL/logs/yolo_all_recipes_sample_rate_2/runs/2023-10-25_12-08-48/test_activity_preds.mscoco.json"
+    )
+    """
+    # coco_val = kwcoco.CocoDataset(
+    #     "/data/PTG/cooking/training/activity_classifier/TCN_HPL/logs/yolo_all_recipes_additional_objs_bkgd_sample_rate_2/runs/2023-11-02_00-52-03/val_activity_preds_epoch43.mscoco.json"
+    # )
+    coco_test = kwcoco.CocoDataset(
+        "/data/users/peri.akiva/PTG/medical/training/activity_classifier/TCN_HPL/logs/p_r18_feat_v6_with_pose_v3_aug_False_reshuffle_True/runs/2024-03-11_09-44-03/test_activity_preds.mscoco.json"
+    )
+
+    config = {
+        "m2": "/home/local/KHQ/peri.akiva/projects/angel_system/config/tasks/m2.yaml",
+        "m3": "/home/local/KHQ/peri.akiva/projects/angel_system/config/tasks/m3.yaml",
+        "m5": "/home/local/KHQ/peri.akiva/projects/angel_system/config/tasks/m5.yaml",
+        "r18": "/home/local/KHQ/peri.akiva/projects/angel_system/config/tasks/r18.yaml",
+    }
```

Comment:
Hard coded paths to the configs? Can this be a parameter or more generic like `angel_system/config/tasks/{task}.yaml`